ddtrace/opentracer/span.py

```diff
@@ -1,39 +1,157 @@
+import time
 from opentracing import Span as OpenTracingSpan
+from ddtrace.span import Span as DatadogSpan
+from ddtrace.ext import errors
+
+
+class SpanLogRecord(object):
+    """A representation of a log record."""
+
+    slots = ['record', 'timestamp']
+
+    def __init__(self, key_values, timestamp=None):
+        self.timestamp = timestamp or time.time()
+        self.record = key_values
+
+
+class SpanLog(object):
+    """A collection of log records."""
+
+    slots = ['records']
+
+    def __init__(self):
+        self.records = []
+
+    def add_record(self, key_values, timestamp=None):
+        self.records.append(SpanLogRecord(key_values, timestamp))
+
+    def __len__(self):
+        return len(self.records)
+
+    def __getitem__(self, key):
+        if type(key) is int and key < len(self):
+            return self.records[key]
+        return None
 
 
 class Span(OpenTracingSpan):
     """Datadog implementation of :class:`opentracing.Span`"""
 
-    def __init__(self, context, tracer, operation_name, tags=None,
-                 start_time=None):
-        pass
+    def __init__(self, tracer, context, operation_name):
+        super(Span, self).__init__(tracer, context)
+
+        # use a datadog span
+        self._span = DatadogSpan(tracer._tracer, operation_name, context=context._context)
```

Comment:
It would help readability to explicitly label this attribute as a Datadog span (`_dd_span` maybe?).